src/combinatorics/set_family/lym.lean

```diff
@@ -0,0 +1,251 @@
+def lym_below (𝒜 : finset (finset α)) : finset (finset α × finset α) :=
+(∂𝒜).sup $ λ t, tᶜ.image $ λ a, (insert a t, t)
+
+/-- For each `t ∈ ∂𝒜`, there are `card α - r + 1` choices for what to add to it to make an element
+of `lym_below`. -/
+lemma _root_.set.sized.card_lym_below (h𝒜 : (𝒜 : set (finset α)).sized r) :
+  (lym_below 𝒜).card = (∂𝒜).card * (fintype.card α - (r - 1)) :=
+begin
+  rw [lym_below, sup_eq_bUnion, card_bUnion],
+  { refine sum_const_nat (λ s hs, _),
+    rw [card_image_of_inj_on, card_compl, h𝒜.shadow hs],
+    intros a ha b hb h,
+    injection h with hab,
+    have q := mem_insert_self a s,
+    rw [hab, mem_insert] at q,
+    exact q.resolve_right (mem_sdiff.1 ha).2 },
+  intros s hs t ht hst,
+  rw disjoint_left,
+  simp_rw [mem_image, not_exists, exists_prop, mem_compl, exists_imp_distrib, prod.forall,
+    prod.mk.inj_iff, and_imp, not_and],
+  rintro _ b i hi rfl rfl j hj k,
+  exact hst.symm,
+end
+
+lemma lym_above_subset_lym_below : lym_above 𝒜 ⊆ lym_below 𝒜 :=
+begin
+  rintro ⟨s, t⟩,
+  simp only [lym_above, lym_below, mem_sup, mem_shadow_iff, true_and, and_imp,
+    exists_prop, mem_sdiff, mem_image, prod.mk.inj_iff, mem_univ, exists_imp_distrib],
+  rintro s hs a hx rfl rfl,
+  exact ⟨s.erase a, ⟨s, hs, a, hx, rfl⟩, a, mem_compl.2 $ not_mem_erase _ _, insert_erase hx, rfl⟩,
+end
+
+/-- The downward **local LYM inequality**. `𝒜` takes up less of `α^(r)` (the finsets of card `r`)
+than `∂𝒜` takes up of `α^(r - 1)`. -/
+lemma local_lym (hr : r ≠ 0) (h𝒜 : (𝒜 : set (finset α)).sized r) :
+  (𝒜.card : 𝕜) / (fintype.card α).choose r ≤ (∂𝒜).card / (fintype.card α).choose (r - 1) :=
+begin
+  obtain hr' | hr' := lt_or_le (fintype.card α) r,
+  { rw [choose_eq_zero_of_lt hr', cast_zero, div_zero],
+    exact div_nonneg (cast_nonneg _) (cast_nonneg _) },
+  { apply lym_aux hr hr',
+    rw [←h𝒜.card_lym_above, ←tsub_tsub_assoc hr' (pos_iff_ne_zero.2 hr), ←h𝒜.card_lym_below],
+    exact card_le_of_subset lym_above_subset_lym_below }
+end
+
+end local_lym
+
+/-! ### LYM inequality -/
+
+section lym
+variables [fintype α]
+
+/-- An inductive definition, from the top down. `falling 𝒜 k` is all the sets with cardinality
+`card α - k` which are a subset of something in `𝒜`. -/
+def falling [decidable_eq α] (𝒜 : finset (finset α)) : Π (k : ℕ), finset (finset α)
+| 0       := 𝒜 # (fintype.card α)
+| (k + 1) := 𝒜 # (fintype.card α - (k + 1)) ∪ ∂ (falling k)
+
+lemma sized_falling [decidable_eq α] (𝒜 : finset (finset α)) (k : ℕ) :
+  (falling 𝒜 k : set (finset α)).sized (fintype.card α - k) :=
+begin
+  induction k with k ih,
+  { exact sized_slice },
+  { rw [falling, coe_union],
+    exact set.sized_union.2 ⟨sized_slice, ih.shadow⟩ }
+end
+
+lemma not_subset_of_mem_slice_of_mem_shadow_falling [decidable_eq α] {𝒜 : finset (finset α)}
+  {r k : ℕ} (hk : k ≤ fintype.card α) (hr : r < k) (h𝒜 : is_antichain (⊆) (𝒜 : set (finset α)))
+  {s t : finset α} (hs : s ∈ 𝒜 # (fintype.card α - k)) (ht : t ∈ ∂ (falling 𝒜 r)) :
+  ¬ s ⊆ t :=
```

Comment:
This lemma is a bit difficult to parse. Maybe a better lemma would be that if `t ∈ falling 𝒜 r` then `t ⊆ t'` for some `t' ∈ 𝒜` (as is hinted in the docstring above). This lemma is easier to understand (has much fewer parameters) and seems to be what you need to prove disjointness later?